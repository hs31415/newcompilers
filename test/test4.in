char num1 = '1', num2 = '2';
for (integer i = 0; i < 2; i = i + 1) {
  write(num1);
  write(num2);
}