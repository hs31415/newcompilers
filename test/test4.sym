KEYWORD function
KEYWORD read
KEYWORD write
KEYWORD return
KEYWORD if
KEYWORD else
KEYWORD for
KEYWORD repeat
KEYWORD until
KEYWORD do
KEYWORD while
KEYWORD integer
KEYWORD double
KEYWORD char
IDENTIFIER num1
IDENTIFIER num2
IDENTIFIER i
