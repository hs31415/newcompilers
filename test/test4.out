 <char>
<id, num1>
 <=>
 <'1'>
 <,>
<id, num2>
 <=>
 <'2'>
 <;>
 <for>
 <(>
 <integer>
<id, i>
 <=>
<NUM, 0>
 <;>
<id, i>
 <<>
<INT, 2>
 <;>
<id, i>
 <=>
<id, i>
 <+>
<INT, 1>
 <)>
 <{>
 <write>
 <(>
<id, num1>
 <)>
 <;>
 <write>
 <(>
<id, num2>
 <)>
 <;>
 <}>
